src/x/net/http/errors.go

```diff
@@ -25,10 +25,19 @@ import (
 	"encoding/json"
 	"errors"
 	"net/http"
+	"sync"
 
 	xerrors "github.com/m3db/m3/src/x/errors"
 )
 
+// ErrorRewriteFn is a function for rewriting response error.
+type ErrorRewriteFn func(error) error
+
+var (
+	_errorRewriteFn     ErrorRewriteFn = func(err error) error { return err }
+	_errorRewriteFnLock sync.Mutex
```

Comment:
We tend not to use underscores for package local variables. Just use a non-exported (lowercase) letter to begin with.

e.g.

```go
errorRewriteFn
errorRewriteFnLock
```